unstructured/partition/utils/ocr_models/ocr_interface.py

```diff
@@ -0,0 +1,120 @@
+from abc import ABC, abstractmethod
+import functools
+import importlib
+from typing import Any, List, Optional, cast
+
+from PIL import Image as PILImage
+from unstructured_inference.inference.elements import TextRegion
+from unstructured_inference.inference.layoutelement import (
+    LayoutElement,
+    partition_groups_from_regions,
+)
+
+from unstructured.documents.elements import ElementType
+
+
+class OCRAgent(ABC):
+    def __init__(self):
+        self.agent = self.load_agent()
+
+    @abstractmethod
+    def load_agent(self, language: str) -> Any:
+        pass
+
+    @abstractmethod
+    def is_text_sorted(self) -> bool:
+        pass
+
+    @abstractmethod
+    def get_text_from_image(self, image: PILImage, ocr_languages: str = "eng") -> str:
+        pass
+
+    @abstractmethod
+    def get_layout_from_image(
+        self, image: PILImage, ocr_languages: str = "eng"
+    ) -> List[TextRegion]:
+        pass
+
+    @abstractmethod
+    def get_layout_elements_from_image(
+        self, image: PILImage, ocr_languages: str = "eng"
+    ) -> List[LayoutElement]:
+        pass
+
+    @staticmethod
+    @functools.lru_cache(maxsize=None)
```

Comment:
My guess is lru cache here is to avoid loading the module multiple times? It does raise a memory lifespan concern: lru cache is not easy to clean; if the python runtime's lifespan and ocr module is the same it is not a big issue; but if the python runtime is supposed to handle followup tasks that might be memory hungry (therefore need cache like this to be released) it can get messy.
One way we can deal with this is to make the `OCRAgent` a singleton pattern https://python-patterns.guide/gang-of-four/singleton/, which doesn't leave cache hidden. Often the memory issue is taken care of by `gc` in this case.

Reply:
Yes, the reason I went with this pattern and the one you mentioned is that I got inspiration from code already here: https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/partition/utils/ocr_models/paddle_ocr.py#L8. But if you confirm you would like me to change it I'm happy to do it, just let me know.

Reply:
actually since this is an existing pattern lets get this in first; and refactor that pattern in a future PR